common4j/src/main/com/microsoft/identity/common/java/cache/AbstractAccountCredentialCache.java

```
@@ -263,24 +260,27 @@ protected List<Credential> getCredentialsFilteredByInternal(@NonNull final List<
                                     || authScheme.equalsIgnoreCase(PopAuthenticationSchemeInternal.SCHEME_POP)
                     );
                 } else {
-                    matches = matches && authScheme.equalsIgnoreCase(atType);
+                    matches = authScheme.equalsIgnoreCase(atType);
                 }
+                if (!matches) continue;
             }
 
             if(mustMatchOnKid && credential instanceof AccessTokenRecord) {
                 final AccessTokenRecord accessToken = (AccessTokenRecord) credential;
-                matches = matches && kid.equalsIgnoreCase(accessToken.getKid());
+                matches = kid.equalsIgnoreCase(accessToken.getKid());
+                if (!matches) continue;
             }
 
             if (mustMatchOnRequestedClaims || mustMatchExactClaims) {
                 if (credential instanceof AccessTokenRecord) {
                     final AccessTokenRecord accessToken = (AccessTokenRecord) credential;
                     if(!(mustMatchExactClaims && StringUtil.isNullOrEmpty(requestedClaims) && StringUtil.isNullOrEmpty(accessToken.getRequestedClaims()))) {
-                        matches = matches && StringUtil.equalsIgnoreCaseTrimBoth(requestedClaims, accessToken.getRequestedClaims());
+                        matches = StringUtil.equalsIgnoreCaseTrimBoth(requestedClaims, accessToken.getRequestedClaims());
                     }
                 } else {
                     Logger.verbose(TAG, "Query specified requested_claims-match, but attempted to match with non-AT credential type.");
                 }
+                if (!matches) continue;
             }
 
             if (matches) {
```

We don't need to check matches here I think because we're failing fast everywhere else right?

---

Updated